pkg/kubelet/network/plugins.go

```diff
@@ -47,6 +48,30 @@ type NetworkPlugin interface {
 
 	// TearDownPod is the method called before a pod's infra container will be deleted
 	TearDownPod(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) error
+
+	// Status is the method called to obtain the ipv4 and ipv6 addresses of the container
+	Status(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) (*PodNetworkStatus, error)
+}
+
+// InterfaceInfo is a structure to store network interface of a pod along with its associated IP addresses
+type InterfaceInfo struct {
+	NetworkInterface net.Interface `json:"networkInterface" description:"A network interface on the pod, e.g. eth1(index:1,mtu:1500,name:eth1..)"`
+	IPAddresses      []net.IP      `json:"ipAddresses" description:"List of IP Addresses associated with the given interface"`
+}
+
+// PodNetworkStatus stores the network status of a pod (currently just the network addresses and interfaces)
+// This struct represents version "v1"
+type PodNetworkStatus struct {
+	api.TypeMeta `json:",inline"`
+
+	// IPAddress is the primary ipv4/ipv6 address of the pod. Among other things it is the address that -
+	//   - kube expects to be reachable across the cluster
+	//   - service endpoints are constructed with
+	//   - will be reported in the PodStatus.PodIP field (will override the IP reported by docker)
+	IPAddress net.IP `json:"ipAddress" description:"Primary address of the pod"`
```

Comment:
can we just call this IP - more consistent with the other API objects.  sorry :)